### DISCUSSION TITLE
Redirect cv page to pdf

### DISCUSSION THREAD

**Discussion by gautamigolani** [alshedivat/al-folio#806] [Q&A] [answered]
Hi! I have updated my website from the previous version and would like to redirect the cv page to a pdf of my cv instead of displaying details on the page itself. Could someone advise how this can be done?

**Comment by adityamwagh** [comment]
You'll need to modify the header html to point to your CV's URL using an anchor tag. :)

**Comment by jxpeng98** [comment]
You can add a file named ``redirected.html`` in folder ``_layouts`` with codes below:
```html
<!DOCTYPE html>
<html>
<head>
    <link rel="canonical" href="{{ page.redirect_to }}"/>
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <meta http-equiv="refresh" content="0;url={{ page.redirect_to }}" />
</head>
<body>
<h1>Redirecting...</h1>
<a href="{{ page.redirect_to }}">Click here if you are not redirected.<a>
    <script>location='{{ page.redirect_to }}'</script>
</body>
</html>
```

and replace the ``cv.md`` in folder ``_pages`` with the below code if your cv is stored in a cloud drive:
```md
---
layout: redirected
permalink: /cv/
redirect_to: https://your.pdf.location
---
```

I think this can solve your question.

**Reply by posmikdc** [reply]
Very nice, this is perfect for me. I ended up slightly changing the yaml header as such: 

```
---
layout: redirected
permalink: /cv/
title: cv
nav: true
nav_order: 4
redirect_to: https://linktomyrepository.com
---
```

That way I can directly link to the repo that contains the most updated version of my CV at all times! If it is no bother, do you know how one could modify the `redirected.html` to open a new tab in the browser? Thanks.

**Comment by sanjit-bhat** [comment, marked as answer]
You could also modify `_layouts/default.html` to include the following:
```html
  <!-- Head -->
  <head>
  {%- if page.redirect -%}
    <meta http-equiv="refresh" content="3; url={{ site.baseurl }}/" />
  {%- endif -%}
  {%- if page.redirect_to -%}
    <meta http-equiv="refresh" content="0; url={{ page.redirect_to }}/" />
  {%- endif -%}
  {% include head.html %}
  </head>
```

Then, your `_pages/cv.md` would look like:
```markdown
---
layout: default
permalink: /cv/
title: cv
nav: true
nav_order: 4
redirect_to: /assets/pdf/your-cv.pdf
---
```

Since this solution re-uses the `default.html` page, it benefits from still including things like analytics scripts for the cv page.

**Comment by marctang** [comment]
In case it is helpful, you can also have the pdf embedded in the webpage. Like this you avoid that people leave the website when reading the pdf. The solution can be found at https://stackoverflow.com/questions/46105960/inserting-my-pdf-in-my-blogs

**Reply by wenxie18** [reply]
Hi, could you please share how to edit related files to embed the cv pdf? Thank you.

**Reply by marctang** [reply]
Sure :-) 

There are three changes to make:

1) put the pdf file of your CV in assets > pdf 
For example, for me it is: assets > pdf > MarcTang_CV_eng.pdf

2) change the layout file
For me, I changed the _layouts > cv.html file to the following:
```
---
layout: default
---
<!-- _layouts/cv.html -->
        <div class="post">

	<object data="../assets/pdf/MarcTang_CV_eng.pdf#pagemode=none" width="750" height="1000" type='application/pdf'></object>

        </div>
```


3) Finally, you change the _pages > cv.md file as well.
For me, the content of the cv.md file is as follows. I think it does not change much here except the title of the page (e.g., the cv_pdf thing does not apply anymore since I changed the layout file in 2).
```
---
layout: cv
permalink: /cv/
title: cv
nav: true
nav_order: 3
cv_pdf: example_pdf.pdf
---
```
If needed, you can see these raw files at the folders of my page: https://github.com/marctang/marctang.github.io

**Reply by wenxie18** [reply]
Thank you so much Marc! It works for me.

**Reply by mac230** [reply]
This is very helpful and most appreciated.  After making @marctang's modifications, my cv pdf was shifted to the right.  In case anyone else faces the same issue, I removed the line "sidebar: left" from "cv.md" in the "_pages" directory to fix this.  The full file is shown below:  
```
---
layout: cv
permalink: /cv/
classes: wide
title: CV
nav: true
nav_order: 1
cv_pdf: cv_mahlon_collins.pdf
toc:
sidebar: left  # <- remove this line to center the pdf
---
```
